adal/src/main/java/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -141,28 +145,77 @@ public AuthenticationContext(Context appContext, String authority, boolean valid
      * @param tokenCacheStore Cache {@link ITokenCacheStore} used to store
      *                        tokens. Set to null if you don't want cache.
      */
-    public AuthenticationContext(Context appContext, String authority,
-                                 ITokenCacheStore tokenCacheStore) {
+    public AuthenticationContext(@NonNull final Context appContext,
+                                 @NonNull final String authority,
+                                 @Nullable final ITokenCacheStore tokenCacheStore) {
         initialize(appContext, authority, tokenCacheStore, true, false);
     }
 
-    private void initialize(Context appContext, String authority, ITokenCacheStore tokenCacheStore,
-                            boolean validateAuthority, boolean defaultCache) {
+    private void initialize(@NonNull final Context appContext,
+                            @NonNull final String authority,
+                            @Nullable final ITokenCacheStore tokenCacheStore,
+                            boolean validateAuthority,
+                            boolean defaultCache) {
         if (appContext == null) {
             throw new IllegalArgumentException("appContext");
         }
+
         if (authority == null) {
             throw new IllegalArgumentException("authority");
         }
+
         mBrokerProxy = new BrokerProxy(appContext);
+
         if (!defaultCache && !mBrokerProxy.canUseLocalCache(authority)) {
             throw new UnsupportedOperationException("Local cache is not supported for broker usage");
         }
+
         mContext = appContext;
         checkInternetPermission();
         mAuthority = extractAuthority(authority);
         mValidateAuthority = validateAuthority;
-        mTokenCacheStore = tokenCacheStore;
+
+        if (null != tokenCacheStore) {
+            mTokenCacheStore = wrapCache(tokenCacheStore);
+        }
+    }
+
+    private ITokenCacheStore wrapCache(@NonNull final ITokenCacheStore originalCache) {
```

Comment:
Do we need to do this? only `removeAll` seems to be doing something special. may be I'm missing something here

Reply:
ADAL supports a custom cache implementation. This is to ensure that this function is still called even if the developer is using their own cache implementation (see constructor where `ITokenCacheStore` can be supplied)